pages/api/super-search.ts

```diff
@@ -0,0 +1,143 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { loadUser } from '@lib/api/middlewares'
+import { Request, SuperSearchResponse } from '@lib/types'
+import { connect } from '@lib/api/db'
+import { findAll } from '@lib/api/entities/superSearch'
+import methods from '@lib/api/api-helpers/methods'
+import runMiddleware from '@lib/api/api-helpers/run-middleware'
+import hasPermission from '@lib/permissions/has-permission'
+import config from '@lib/config'
+import { ROLES } from '@lib/constants'
+
+const {
+  superSearch: { entities, enabled },
+} = config
+
+export default async (
+  request: NextApiRequest,
+  response: NextApiResponse<{ data: SuperSearchResponse[] } | { error: string }>
+) => {
+  try {
+    // Connect to database
+    await connect()
+  } catch (e) {
+    return response.status(500).json({
+      error: e.message,
+    })
+  }
+
+  try {
+    await runMiddleware(request, response, loadUser)
+  } catch (e) {
+    return response.status(500).json({
+      error: e.message,
+    })
+  }
+
+  const { currentUser } = request as Request
+
+  if (
+    !hasPermission(currentUser, ['superSearch.read', 'user.admin']) ||
+    !enabled
+  ) {
+    return response.status(401).json({
+      error: 'Access forbidden.',
+    })
+  }
+
+  methods(request, response, {
+    get: getEntities,
+  })
+}
+
+async function getEntities(
+  request: Request & { query: { query: string } },
+  response: NextApiResponse<{ data: SuperSearchResponse[] }>
+): Promise<void> {
+  const {
+    query: { query },
+    currentUser,
+  } = request
+
+  if (!query) {
+    return response.json({ data: [] })
+  }
+
+  const filteredEntities = []
+
+  if (currentUser) {
+    const { roles: userRoles } = currentUser
+    // TODO: maybe it can be done in the better way
+    filteredEntities.push(
+      ...entities.reduce((acc, entity) => {
+        const { permissions: roles, type } = entity
+
+        const canSee = roles.some(
+          (role) =>
```

Comment:
please, use the "hasPermission" utility , do not implement a new way to check for permissions